src/main/java/io/stargate/sgv2/jsonapi/service/shredding/Shredder.java

```diff
@@ -75,7 +75,7 @@ public WritableShreddedDocument shred(JsonNode doc, UUID txId, DocumentProjector
     // Now that we have the traversable document, verify it does not violate
     // structural limits, before serializing.
     // (note: value validation has to wait until no-indexing projection is applied)
-    new StructuralValidator(documentLimits).validate(docWithId);
+    new FullDocValidator(documentLimits).validate(docWithId);
```

Comment:
Changed name to better describe now that structural validation is split between full document (before trimming non-indexable fields) and indexable-only document.